### PR TITLE
Fix for issue  #58050

### PR DESCRIPTION
**Pull request by asettouf**
Hi,

a quick PR to mention in the compiler error message that `?` is a macro operator, as according to issue #58050 

It passed `python x.py test src/tools/tidy`  locally, as well as the recommendation to run `/x.py test src/test/ui --stage 1 --bless`.

Let me know if anything else is needed.

### CODE REVIEW

File: src/libsyntax/ext/tt/quoted.rs [hunk 1]
```diff
@@ -474,11 +474,13 @@ where
...
                 // (N.B. We need to advance the input iterator.)
                 match parse_kleene_op(input, span) {
-                    // #2 is `?`, which is not allowed as a Kleene op in 2015 edition.
+                    // #2 is `?`, which is not allowed as a Kleene op in 2015 edition,
+                    //but is allowed in the 2018 edition
                     Ok(Ok((op, op2_span))) if op == KleeneOp::ZeroOrOne => {
                         sess.span_diagnostic
                             .struct_span_err(op2_span, "expected `*` or `+`")
-                            .note("`?` is not a macro repetition operator")
+                            .note("`?` is not a macro repetition operator in the 2015 edition,\
```
Comment (varkor) [inline src/libsyntax/ext/tt/quoted.rs]:
```suggestion
                            .note("`?` is not a macro repetition operator in the 2015 edition, \
```
(Here and in the other places, to ensure there's a space after the comma.

File: src/libsyntax/ext/tt/quoted.rs [hunk 2]
```diff
@@ -474,11 +474,13 @@ where
                 // #1 is a separator and #2 should be a KleepeOp.
                 // (N.B. We need to advance the input iterator.)
                 match parse_kleene_op(input, span) {
-                    // #2 is `?`, which is not allowed as a Kleene op in 2015 edition.
+                    // #2 is `?`, which is not allowed as a Kleene op in 2015 edition,
+                    //but is allowed in the 2018 edition
```
Comment (varkor) [inline src/libsyntax/ext/tt/quoted.rs]:
```suggestion
                    // but is allowed in the 2018 edition.
```

File: src/libsyntax/ext/tt/quoted.rs [hunk 3]
```diff
@@ -506,10 +508,12 @@ where
                     Err(_) => op1_span,
                 }
             } else {
-                // `?` is not allowed as a Kleene op in 2015
+                // `?` is not allowed as a Kleene op in 2015,
+                // but is allowed in the 2018 edition
```
Comment (varkor) [inline src/libsyntax/ext/tt/quoted.rs]:
```suggestion
                // but is allowed in the 2018 edition.
```

File: src/libsyntax/ext/tt/quoted.rs [hunk 4]
```diff
@@ -506,10 +508,12 @@ where
...
                 sess.span_diagnostic
                     .struct_span_err(op1_span, "expected `*` or `+`")
-                    .note("`?` is not a macro repetition operator")
+                    .note("`?` is not a macro repetition operator in the 2015 edition,\
```
Comment (varkor) [inline src/libsyntax/ext/tt/quoted.rs]:
```suggestion
                    .note("`?` is not a macro repetition operator in the 2015 edition, \
```

File: src/libsyntax/ext/tt/quoted.rs [hunk 5]
```diff
@@ -519,11 +523,13 @@ where
 
         // #1 is a separator followed by #2, a KleeneOp
         Ok(Err((tok, span))) => match parse_kleene_op(input, span) {
-            // #2 is a `?`, which is not allowed as a Kleene op in 2015 edition.
+            // #2 is a `?`, which is not allowed as a Kleene op in 2015 edition,
+            // but is allowed in the 2018 edition
```
Comment (varkor) [inline src/libsyntax/ext/tt/quoted.rs]:
```suggestion
            // but is allowed in the 2018 edition.
```

File: src/libsyntax/ext/tt/quoted.rs [hunk 6]
```diff
@@ -519,11 +523,13 @@ where
...
             Ok(Ok((op, op2_span))) if op == KleeneOp::ZeroOrOne => {
                 sess.span_diagnostic
                     .struct_span_err(op2_span, "expected `*` or `+`")
-                    .note("`?` is not a macro repetition operator")
+                    .note("`?` is not a macro repetition operator in the 2015 edition,\
```
Comment (varkor) [inline src/libsyntax/ext/tt/quoted.rs]:
```suggestion
                    .note("`?` is not a macro repetition operator in the 2015 edition, \
```

File: src/libsyntax/ext/tt/quoted.rs [hunk 7]
```diff
@@ -519,11 +523,13 @@ where
...
+                        but accepted in the 2018 edition")
```
Comment (varkor) [inline src/libsyntax/ext/tt/quoted.rs]:
```suggestion
                        but is accepted in the 2018 edition")
```

File: src/libsyntax/ext/tt/quoted.rs [hunk 8]
```diff
@@ -474,11 +474,13 @@ where
...
                 match parse_kleene_op(input, span) {
-                    // #2 is `?`, which is not allowed as a Kleene op in 2015 edition.
+                    // #2 is `?`, which is not allowed as a Kleene op in 2015 edition,
+                    //but is allowed in the 2018 edition
                     Ok(Ok((op, op2_span))) if op == KleeneOp::ZeroOrOne => {
                         sess.span_diagnostic
                             .struct_span_err(op2_span, "expected `*` or `+`")
-                            .note("`?` is not a macro repetition operator")
+                            .note("`?` is not a macro repetition operator in the 2015 edition,\
+                                 but accepted in the 2018 edition")
```
Comment (varkor) [inline src/libsyntax/ext/tt/quoted.rs]:
```suggestion
                                 but is accepted in the 2018 edition")
```

File: src/libsyntax/ext/tt/quoted.rs [hunk 9]
```diff
@@ -506,10 +508,12 @@ where
...
                 }
             } else {
-                // `?` is not allowed as a Kleene op in 2015
+                // `?` is not allowed as a Kleene op in 2015,
+                // but is allowed in the 2018 edition
                 sess.span_diagnostic
                     .struct_span_err(op1_span, "expected `*` or `+`")
-                    .note("`?` is not a macro repetition operator")
+                    .note("`?` is not a macro repetition operator in the 2015 edition,\
+                         but accepted in the 2018 edition")
```
Comment (varkor) [inline src/libsyntax/ext/tt/quoted.rs]:
```suggestion
                         but is accepted in the 2018 edition")
```

**Comment by asettouf** [comment]
@varkor Good suggestions, my bad!

File: src/libsyntax/ext/tt/quoted.rs (hunk 4 above)
Comment (varkor) [inline src/libsyntax/ext/tt/quoted.rs]:
```suggestion
                    .note("`?` is not a macro repetition operator in the 2015 edition, \
```

File: src/libsyntax/ext/tt/quoted.rs (hunk 9 above)
Comment (varkor) [inline src/libsyntax/ext/tt/quoted.rs]:
```suggestion
                         but is accepted in the 2018 edition")
```

File: src/libsyntax/ext/tt/quoted.rs (hunk 6 above)
Comment (varkor) [inline src/libsyntax/ext/tt/quoted.rs]:
```suggestion
                    .note("`?` is not a macro repetition operator in the 2015 edition, \
```

File: src/libsyntax/ext/tt/quoted.rs (hunk 7 above)
Comment (varkor) [inline src/libsyntax/ext/tt/quoted.rs]:
```suggestion
                        but is accepted in the 2018 edition")
```

File: src/test/ui/macros/macro-at-most-once-rep-2015-ques-rep.stderr [hunk 10]
```diff
@@ -4,15 +4,15 @@ error: expected `*` or `+`
 LL |     ($(a)?) => {} //~ERROR expected `*` or `+`
    |          ^
    |
-   = note: `?` is not a macro repetition operator
+   = note: `?` is not a macro repetition operator in the 2015 edition,but accepted in the 2018 edition
```
Comment (varkor) [inline src/test/ui/macros/macro-at-most-once-rep-2015-ques-rep.stderr]:
```suggestion
   = note: `?` is not a macro repetition operator in the 2015 edition, but is accepted in the 2018 edition
```

File: src/test/ui/macros/macro-at-most-once-rep-2015-ques-rep.stderr [hunk 11]
```diff
@@ -4,15 +4,15 @@ error: expected `*` or `+`
...
 
 error: expected `*` or `+`
   --> $DIR/macro-at-most-once-rep-2015-ques-rep.rs:10:11
    |
 LL |     ($(a),?) => {} //~ERROR expected `*` or `+`
    |           ^
    |
-   = note: `?` is not a macro repetition operator
+   = note: `?` is not a macro repetition operator in the 2015 edition,but accepted in the 2018 edition
```
Comment (varkor) [inline src/test/ui/macros/macro-at-most-once-rep-2015-ques-rep.stderr]:
```suggestion
   = note: `?` is not a macro repetition operator in the 2015 edition, but is accepted in the 2018 edition
```

**Comment by varkor** [comment]
@asettouf: I should have clarified — the other two messages needed the same changes (I've made specific suggestions now). After that, it looks good to me!

**Comment by michaelwoerister** [comment]
@varkor, do you want to take over the review? You seem to already have taken a close look at this.

**Comment by varkor** [comment]
@asettouf: this looks good now! If you could squash your changes into a single commit, then I'll approve. (Note also that we avoid merge commits in Rust, so if you get any conflicts you should rebase on top of master instead.)

r? @varkor

**Comment by varkor** [comment]
@asettouf: are you able to squash the commits? Everything looks great other than that.

**Comment by asettouf** [comment]
@varkor  Sorry for the delay, it's been squashed. Besides, I made a slight mistake not creating a branch on my repo and then proceeded with merging rust:master into mine, which I just reverted. (that's why in the long message of the commit you can see a revert happening)

**Comment by varkor** [comment]
@asettouf: thanks! That's perfect.

@bors r+

**Pull request merged**